### DISCUSSION TITLE
Drop support of platform `arm/v6`.

### DISCUSSION THREAD

**Discussion by googollee** [photoview/photoview#1051] [General]
While working with `jellyfin-ffmpeg`, I discovered that the `arm/v6` build was incorrectly configured to use the `armhf` architecture. This results in `jellyfin-ffmpeg` with `armhf` failing to install on `debian:bookworm-slim` image with `arm/v6` arch, as shown in [the logs](https://github.com/googollee/photoview/actions/runs/10847238167/job/30101897524#step:7:17564).  The `arm/v7` build installs correctly, also demonstrated in [the logs](https://github.com/googollee/photoview/actions/runs/10847441836/job/30102518194#step:7:10490).

According to [Debian Arm Support](https://www.debian.org/ports/arm/#status), `arm/v6` should use the `armel` architecture.  [Wiki](https://en.wikipedia.org/wiki/ARM_architecture_family#Floating-point_(VFP)) says `arm/v6` optionally relies on the VFPv2 instructions for hardware floating-point support, which is incompatible with the VFPv3 instructions used by `armhf`.  Therefore, installing `jellyfin-ffmpeg` with `armhf` for `arm/v6` is incorrect.

**I propose dropping `arm/v6` support altogether, as it's unlikely to function correctly with the current build system.**

Alternatively, we could utilize the `ffmpeg` package from the Debian repository for `arm/v6` platforms, as `jellyfin-ffmpeg` doesn't currently offer an `armel` binary. However, since `arm/v6` usage appears to be minimal, I recommend discontinuing support to simplify the build process and maintenance.

Are there any users running `photoview` on `arm/v6` devices?

@kkovaletp @viktorstrate What are your thoughts on this?

**Comment by kkovaletp** [comment]
@googollee, in general, I'm fine with dropping the `arm/v6` support. But analysing your findings I think that actually, Photoview might at least partially work on that platform:
Assuming that you refer to the https://github.com/photoview/photoview/blob/master/scripts/install_build_dependencies.sh , the difference is only in 2 dependencies (compiler packages), used to build the code. As our product is relatively small and simple, we might be locky enough not to use (or use in a limited way) libs and functions, which are compiled differently and incompatible for these 2 arch's...

Unfortunately, we don't have a HW to pull and test the image, so these are only assumptions...

**Reply by googollee** [reply]
Both [`install_build_dependencies.sh`](https://github.com/photoview/photoview/blob/master/scripts/install_build_dependencies.sh#L7) and [`set_compiler_env.sh`](https://github.com/photoview/photoview/blob/a6cbfc76f3d4acbf9dd3e5767a34ba319fc4cd56/scripts/set_compiler_env.sh#L70) set `arm/v6` as `armhf`.

It's not only that 2 packages, but also `ffmpeg` and `imagemagick`.
